### PR TITLE
Largest product array

### PR DESCRIPTION
**Pull request by dsnowb**
- [x] file structure
- [x] completed whiteboard
- [x] implemented whiteboard
- [x] completed testing
- [x] updated documentation

### CODE REVIEW

**Comment by dsnowb** [comment]
- [x] file structure
- [x] completed whiteboard
- [x] implemented whiteboard
- [x] completed testing
- [x] updated documentation

File: challenges/largest-product-array/largest_product.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    '''Returns the largest product of adjacent values in an m x n matrix, given m > 1, n > 1'''
+
+    # Check param type and values
+    if type(arr) is not list:
+        raise TypeError('Parameter must of be of type <list>')
+    if not (len(arr[0]) > 1 and len(arr) > 1):
+        raise ValueError('Function expects 2 x 2 list or greater')
+
+    # Initialize our max product
+    max_p = arr[0][0] * arr[1][0]
```
Comment (Woojgh) [inline challenges/largest-product-array/largest_product.py]: So by putting this in you wont have to set your max product to zero but it only allows you to run against certain cases, its not inclusive of all cases.

File: challenges/largest-product-array/largest_product.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            if hb > max_p:
+                max_p = hb
+            if vb > max_p:
+                max_p = vb
+            if da > max_p:
+                max_p = da
+            if db > max_p:
+                max_p = db
+
+            print('ha{} hb{} vb{} da{} db{}'.format(ha,hb,vb,da,db))
```
Comment (Woojgh) [inline challenges/largest-product-array/largest_product.py]: remove prints from final code. Im assuming you had this for personal viewing.

File: challenges/largest-product-array/largest_product.py (hunk 1 above)
Comment (dsnowb) [reply challenges/largest-product-array/largest_product.py]: Are you sure about that? Pretty sure that everything with min dimensionality 2x2 is tested, and anything with less dimensionality is not 2D. Also anything less than 2x2 can't safely have a product initialized to 0.

File: challenges/largest-product-array/largest_product.py (hunk 2 above)
Comment (dsnowb) [reply challenges/largest-product-array/largest_product.py]: Totally good call. Chalk that one up to gotta work less time, heh.

**Pull request merged**